monai/networks/blocks/fcn.py

```diff
@@ -0,0 +1,245 @@
+    """
+
+    def __init__(self, inplanes: int, planes: int, ks: int = 7):
+        """
+        Args:
+            inplanes (int): number of input channels.
+            planes (int)： number of output channels.
+            ks (int): kernel size for one dimension. Defaults to 7.
+        """
+        super(GCN, self).__init__()
+
+        conv2d_type: Type[nn.Conv2d] = Conv[Conv.CONV, 2]
+        self.conv_l1 = conv2d_type(in_channels=inplanes, out_channels=planes, kernel_size=(ks, 1), padding=(ks // 2, 0))
+        self.conv_l2 = conv2d_type(in_channels=planes, out_channels=planes, kernel_size=(1, ks), padding=(0, ks // 2))
+        self.conv_r1 = conv2d_type(in_channels=inplanes, out_channels=planes, kernel_size=(1, ks), padding=(0, ks // 2))
+        self.conv_r2 = conv2d_type(in_channels=planes, out_channels=planes, kernel_size=(ks, 1), padding=(ks // 2, 0))
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            x (torch.Tensor): in shape (batch, inplanes, spatial_1, spatial_2).
+        """
+        x_l = self.conv_l1(x)
+        x_l = self.conv_l2(x_l)
+        x_r = self.conv_r1(x)
+        x_r = self.conv_r2(x_r)
+        x = x_l + x_r
+        return x
+
+
+class Refine(nn.Module):
+    """
+    Simple residual block to refine the details of the activation maps.
+    The code is adapted from lsqshr's original version:
+    https://github.com/lsqshr/AH-Net/blob/master/net2d.py
+    """
+
+    def __init__(self, planes: int):
+        """
+        Args:
+            planes (int): number of input channels.
+        """
+        super(Refine, self).__init__()
+
+        relu_type: Type[nn.ReLU] = Act[Act.RELU]
+        conv2d_type: Type[nn.Conv2d] = Conv[Conv.CONV, 2]
+        norm2d_type: Type[nn.BatchNorm2d] = Norm[Norm.BATCH, 2]
+
+        self.bn = norm2d_type(num_features=planes)
+        self.relu = relu_type(inplace=True)
+        self.conv1 = conv2d_type(in_channels=planes, out_channels=planes, kernel_size=3, padding=1)
+        self.conv2 = conv2d_type(in_channels=planes, out_channels=planes, kernel_size=3, padding=1)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            x (torch.Tensor): in shape (batch, planes, spatial_1, spatial_2).
+        """
+        residual = x
+        x = self.bn(x)
+        x = self.relu(x)
+        x = self.conv1(x)
+        x = self.bn(x)
+        x = self.relu(x)
+        x = self.conv2(x)
+
+        out = residual + x
+        return out
+
+
+class FCN(nn.Module):
+    """
+    2D FCN network with 3 input channels. The small decoder is built
+    with the GCN and Refine modules.
+    The code is adapted from lsqshr's original version:
+    https://github.com/lsqshr/AH-Net/blob/master/net2d.py
+
+    Args:
+        nout (int): number of output channels. Defaults to 1.
+        deter_flag (bool): let deter_flag=True if needs to make the training process deterministic.
```

Comment:
`deter_flag` will only affect the upsampling in this particular module, how about changing `deter_flag` to `upsample_mode` and making comments that some choices of `upsample_mode` couldn't provide reproducibility? it becomes more intuitive and in the future we may extend the upsampling mode (it's the same logic for a few other modules that have `deter_flag`

Reply:
I see, I've updated.